### ISSUE TITLE
Cube settings don’t work

### ISSUE THREAD

**Issue by geremi1** [compiz-reloaded/compiz#104]
In ‘rotate cube’, the zoom do not work properly
In ‘3d windows’ the window depth do not work at all

**Comment by soreau** [comment]
@geremi1 Hi, what version of compiz are you using?

**Comment by geremi1** [comment]
The lastest you’ve created (3days ago)

**Comment by soreau** [comment]
How are you starting compiz?

**Comment by geremi1** [comment]
Yes, with ‘compiz --replace

**Comment by soreau** [comment]
Is libcompizconfig installed? Does compiz show any output about not being able to load ccp?

**Comment by geremi1** [comment]
No, the only case I see an error is ‘topmenu-gtk-‘

**Comment by soreau** [comment]
It sounds like ccp isn't loaded or working for some reason. Possible causes could include wrong permissions on ~/.config/compiz/compizconfig/Default.ini, automatic plugin sorting not being used in ccsm>Preferences, or lack of libcompizconfig or compizconfig-python being installed. You can also try Reset to Defaults in ccsm>Preferences and reconfigure ccsm.

**Comment by geremi1** [comment]
Ok, is compiz-reloaded can be install beside compiz-core (unity)?

**Comment by soreau** [comment]
No, if you install compiz-reloaded, you should first remove all compiz and libdecoration 0.9.x packages.

**Comment by geremi1** [comment]
Do you know why the package libwnck-dev cannot be installed on a live usb ubuntu 16.04?

**Comment by geremi1** [comment]
It is required by compiz-reloaded

**Comment by soreau** [comment]
Maybe need apt-get update first? not too sure

**Comment by geremi1** [comment]
After a complete fresh installation of Xubuntu 16.04.3 LTS 64-bit, and Compiz-Reloaded 0.8.14, the viewport distance, curve strengh in expo mode, and again the 3d window’s depth, the zoom distance in Rotate cube aren’t working properly, only the default or the most extreme work. Any idea why?

**Comment by soreau** [comment]
It sounds like compiz isn't loading ccp plugin. Make sure libcompizconfig and compizconfig-python are correctly installed, then try `compiz --replace ccp`.

**Comment by soreau** [comment]
I tested the settings you mentioned and they work fine on git master 0.8.15. I'm not aware of anything that was done to cause or fix this problem and you are the first to report it. Are you running compiz as root or have you at some point? There might be a permissions problem on ~/.config/compiz/compizconfig/Default.ini so check that it is rw and owned by your user.

**Comment by geremi1** [comment]
I’ll answer you tomorrow

**Comment by geremi1** [comment]
Here what I have as configuration file (configIssue#104.profile) [config-Issue#104.zip](https://github.com/compiz-reloaded/compiz/files/1670422/config-Issue.104.zip)

Here is a picture of what the terminal shows me when I do the command you mentioned before. [screenshot](https://user-images.githubusercontent.com/31183791/35475302-ea5ab1a8-0369-11e8-9e77-232d5a00ed40.png)

Here is a picture of what I think is the error about (the grid plugin). [screenshot](https://user-images.githubusercontent.com/31183791/35475311-08cd01ae-036a-11e8-98eb-cc2af0c64743.png)

For installing, I use this site: http://blog.northfield.ws/compiz-release-announcement-0-8-14/

**Comment by soreau** [comment]
Those error messages should be safe to ignore.

Did you try Reset to Defaults in ccsm>Preferences and reconfiguring ccsm (manually, not by importing profile)? You should probably do it while compiz isn't running. Not sure this will help but I'm not really sure what the problem is here. Also what does `ls -l ~/.config/compiz/compizconfig/Default.ini` say?

**Comment by geremi1** [comment]
It says:” -rw-rw-r-- 1 compiztest compiztest 80 jan 23 14:51 /home........ (the file)”

**Comment by soreau** [comment]
Well that part looks ok at least.

**Comment by geremi1** [comment]
And after I launch it with “compiz —replace ccp”, and modify a single parameter, it prompts:
A handler is already registered for /org/freedesktop/compiz/***

Where “***” stands for obs, text, trailfocus, clone, stackswitch, resizeinfo, cube, expo, shift, ezoom, ring, showmouse, scale, rotate, scaleaddon.

**Comment by soreau** [comment]
Yes, that is normal, I think it comes from dbus plugin.

**Comment by geremi1** [comment]
I tested, and yes, it is

**Comment by soreau** [comment]
If you change one of the options that aren't working correctly to a specified value (not default or max) and restart compiz does it take effect at all?

**Comment by geremi1** [comment]
No, it even rests them all

**Comment by geremi1** [comment]
To 0.0000 value, not the default one

**Comment by soreau** [comment]
Are you able to upgrade to latest git to see if this is happening? If you are using the installation scripts, from the scripts directory, run `./uninstall && git checkout master && ./go`

**Comment by geremi1** [comment]
The script has finished, what I do?

**Comment by soreau** [comment]
Restart ccsm and compiz.

**Comment by soreau** [comment]
Try without importing a profile, enable basic plugins (move, resize, place, decoration) and the plugins that have the problem (rotate, 3D). If you have already imported a profile, use Reset to Defaults button and then enable the plugins above and start compiz.

**Comment by geremi1** [comment]
Do same as before.

BUT: the terminal says: “compiz (cube) - Warn: Failed to load slide: freedesktop”

**Comment by soreau** [comment]
That message is because png plugin isn't loaded. Can you try removing ~/.cache/compizconfig/*.pb files and restarting compiz and ccsm?

**Comment by geremi1** [comment]
It’ll be tomorrow, I cannot touch my computer anymore till then

**Comment by geremi1** [comment]
THANKS YOU SO MUCH!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
There’s only one thing, the “blur windows” isn’t working properly

**Comment by soreau** [comment]
Is that to say removing the .pb files helped make the 'broken' settings start working again?

What's wrong with blur windows?

**Comment by geremi1** [comment]
Yes, exactly.

It doesn’t do anything, and aplha is really weird

**Comment by soreau** [comment]
Can you explain what you mean by 'really weird'? A screenshot would help. Also what is the output of `glxinfo|grep renderer`?

**Comment by geremi1** [comment]
Here the [screenshot](https://user-images.githubusercontent.com/31183791/35486197-8c7a6fc2-0438-11e8-9813-6e8648584939.png)

Here the [screenshot](https://user-images.githubusercontent.com/31183791/35486229-00c6c010-0439-11e8-9cab-e1a1caa9d808.png) of what’s the command prints

**Comment by soreau** [comment]
There is an apparent bug in the R600 driver for your card. This can be worked around by starting compiz with `R600_DEBUG=sbsafemath compiz --replace`

**Comment by geremi1** [comment]
Always?

**Comment by soreau** [comment]
Yes

**Comment by geremi1** [comment]
Ok, without ccp?

**Comment by soreau** [comment]
Yes, actually when specifying no plugins, it automatically loads ccp, move, resize, decoration and place so these can't be disabled in ccsm. If you want to be able to disable any of these plugins, you should pass ccp.

**Comment by geremi1** [comment]
Ok, thanks so much, I tell you when something get wrong or if I had a new feature request :)

**Comment by soreau** [comment]
No problem, thanks for verifying my suspicion that these .pb files can sometimes cause strange issues. I'm thinking we should have ccsm delete them when clicking the Reset the Defaults button at least. Perhaps the problem arises when importing a profile.
